contracts/multisig/lib.rs

```diff
@@ -0,0 +1,303 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_lang as ink;
+
+#[ink::contract(env = serai_extension::SeraiEnvironment)]
+mod multisig {
+  use scale::Encode;
+
+  use ink_storage::{traits::SpreadAllocate, Mapping};
+  use ink_env::{hash::Blake2x256, hash_encoded};
+
+  /// A contract which tracks the current multisig keys.
+  #[ink(storage)]
+  #[derive(SpreadAllocate)]
+  pub struct Multisig {
+    /// Validator set currently holding the multisig.
+    validator_set: [u8; 32],
+    /// Mapping from a curve's index to the multisig's current public key for it.
+    // This is a mapping due to ink's eager loading. Considering we're right now only considering
+    // secp256k1 and Ed25519, it may be notably more efficient to use a Vec here.
+    keys: Mapping<u8, Vec<u8>>,
+    /// Voter + Keys -> Voted already or not
+    voted: Mapping<(AccountId, [u8; 32]), ()>,
+    /// Validator Set + Keys -> Vote Count
+    votes: Mapping<([u8; 32], [u8; 32]), u16>,
```

Comment:
what are the `keys` representing here?

Reply:
The hash of all multisig keys being proposed. Will add a comment clarifying.